x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -736,23 +713,46 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
         });
     };
 
-    function findFieldType(index: string) {
-      getFieldTypeFromMapping(index, categorizationFieldName)
-        .then((resp: string) => {
-          if (resp !== '') {
-            createAndOpenUrl(datafeedIndices.join(), resp);
-          } else {
-            i++;
-            if (i < datafeedIndices.length) {
-              findFieldType(datafeedIndices[i]);
-            } else {
-              error();
-            }
-          }
+    // Find the type of the categorization field i.e. text (preferred) or keyword.
+    // Uses the first matching field found in the list of indices in the datafeed_config.
+    let fieldType;
+
+    for (const index of datafeedIndices) {
+      const dataView = (await data.dataViews.find(index)).find(
+        (dv) => dv.getIndexPattern() === index
+      );
+
+      if (!dataView) {
```

Comment:
Testing this, I discovered that we no longer gracefully handle (on `main`) the case where there is no longer a matching data view for the indices used in the datafeed. We now just silently fail, where the menu item used to be disabled as implemented in https://github.com/elastic/kibana/pull/125090.

With ad hoc data views, we could actually enhance the behavior where a data view doesn't exist by enabling the link and let Discover create a temporary data view:

<img width="434" alt="Screenshot 2024-05-07 at 16 29 14" src="https://github.com/elastic/kibana/assets/7405507/db4a0d49-f4ac-4e4a-b7d1-b208b9027e92">

Reply:
I took an attempt at fixing this. The problem is that the way we link to Discover is quite different for the `View in Discover` and the `View examples` link. For the first one the temporary data view works because we don't open the link in a new window. To align this would require quite a refactor of the whole `links_menu.tsx` file. I tried to keep changes to a minimum and tried to restore the previous behavior until we find the time to align: For now `View in Discover` will work with temporary data views whereas the `View examples` link will be disabled with a tooltip explaining that the data view doesn't exist. Updated in [0db6610](https://github.com/elastic/kibana/pull/182588/commits/0db66102469f7e89a012f6fa5f319c30f6e994fa).